### PR TITLE
Remove some race conditions and add more DisposeOnce

### PR DESCRIPTION
**Pull request by jbayardo**

### CODE REVIEW

File: src/Raven.Server/Documents/Indexes/IndexMerging/IndexMerger.cs [hunk 1]
```diff
@@ -352,7 +352,6 @@ private static void MergeSelectExpressionsAndFields(MergeProposal mergeProposal,
 
                 mergeSuggestion.CanMerge.Add(curProposedData.IndexName);
                 DataDictionaryMerge(mergeSuggestion.MergedIndex.Fields, curProposedData.Index.Fields);
-                mergeSuggestion.MergedIndex.Fields = mergeSuggestion.MergedIndex.Fields;
```
Comment (ayende) [inline src/Raven.Server/Documents/Indexes/IndexMerging/IndexMerger.cs:355]: Why remove this?

File: src/Raven.Server/Documents/Indexes/Persistence/Lucene/LuceneSuggestionIndexWriter.cs [hunk 2]
```diff
@@ -263,7 +263,7 @@ private void DisposeIndexWriter(bool waitForMerges = true)
 
             try
             {
-                writer?.Analyzer.Close();
```
Comment (ayende) [inline src/Raven.Server/Documents/Indexes/Persistence/Lucene/LuceneSuggestionIndexWriter.cs:266]: Can this have been concurrently changed here?

File: src/Raven.Server/Documents/Indexes/IndexMerging/IndexMerger.cs (hunk 1 above)
Comment (jbayardo) [reply src/Raven.Server/Documents/Indexes/IndexMerging/IndexMerger.cs:355]: Must have been Resharper. It is not useful though, so it shouldn't do any harm

File: src/Raven.Server/Documents/Indexes/Persistence/Lucene/LuceneSuggestionIndexWriter.cs (hunk 2 above)
Comment (jbayardo) [reply src/Raven.Server/Documents/Indexes/Persistence/Lucene/LuceneSuggestionIndexWriter.cs:266]: If that code runs concurrently, we have a race condition at

            if (_indexWriter == null)
                 return;

So the assumption is that it doesn't

**Pull request merged**